x/minfee/module.go

```diff
@@ -0,0 +1,145 @@
+	"github.com/spf13/cobra"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkmodule "github.com/cosmos/cosmos-sdk/types/module"
+	params "github.com/cosmos/cosmos-sdk/x/params/keeper"
+)
+
+var (
+	_ sdkmodule.AppModule      = AppModule{}
+	_ sdkmodule.AppModuleBasic = AppModuleBasic{}
+)
+
+// AppModuleBasic defines the basic application module used by the minfee module.
+type AppModuleBasic struct{}
+
+// RegisterInterfaces registers the module's interfaces with the interface registry.
+func (AppModuleBasic) RegisterInterfaces(_ cdctypes.InterfaceRegistry) {}
+
+// Name returns the minfee module's name.
+func (AppModuleBasic) Name() string {
+	return ModuleName
+}
+
+// RegisterLegacyAminoCodec does nothing. MinFee doesn't use Amino.
+func (AppModuleBasic) RegisterLegacyAminoCodec(*codec.LegacyAmino) {}
+
+// DefaultGenesis returns default genesis state as raw bytes for the minfee module.
+func (AppModuleBasic) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
+	return cdc.MustMarshalJSON(DefaultGenesis())
+}
+
+// ValidateGenesis performs genesis state validation for the minfee module.
+func (AppModuleBasic) ValidateGenesis(cdc codec.JSONCodec, _ client.TxEncodingConfig, bz json.RawMessage) error {
+	var data GenesisState
+	if err := cdc.UnmarshalJSON(bz, &data); err != nil {
+		return fmt.Errorf("failed to unmarshal %s genesis state: %w", ModuleName, err)
+	}
+	return ValidateGenesis(&data)
+}
+
+// RegisterRESTRoutes registers the REST service handlers for the module.
+func (AppModuleBasic) RegisterRESTRoutes(_ client.Context, _ *mux.Router) {}
+
+// RegisterGRPCGatewayRoutes registers the gRPC Gateway routes for the module.
+func (AppModuleBasic) RegisterGRPCGatewayRoutes(_ client.Context, _ *runtime.ServeMux) {}
+
+// GetTxCmd returns the minfee module's root tx command.
+func (a AppModuleBasic) GetTxCmd() *cobra.Command {
+	// Return a dummy command
+	return &cobra.Command{}
+}
+
+// GetQueryCmd returns the a dummy command.
+func (AppModuleBasic) GetQueryCmd() *cobra.Command {
+	// Return a dummy command
+	return &cobra.Command{}
+}
+
+// AppModule implements an application module for the minfee module.
+type AppModule struct {
+	AppModuleBasic
+	paramsKeeper params.Keeper
+}
+
+// NewAppModule creates a new AppModule object
+func NewAppModule(k params.Keeper) AppModule {
+	return AppModule{
+		AppModuleBasic: AppModuleBasic{},
+		paramsKeeper:   k,
+	}
+}
+
+// RegisterInvariants registers the minfee module invariants.
+func (am AppModule) RegisterInvariants(_ sdk.InvariantRegistry) {}
+
+// Route returns the message routing key for the minfee module.
+func (am AppModule) Route() sdk.Route {
+	return sdk.Route{}
+}
+
+// QuerierRoute returns the minfee module's querier route name.
+func (am AppModule) QuerierRoute() string {
+	return ""
```

Comment:
do we need a querier route?

Reply:
No, we're just mocking the rest of the interface for the module. We only really need the module for initialising the genesis

Reply:
yeah thought so but I added it anyway. we could remove it or we could leave it.

Reply:
Sorry Cosmos SDK docs are awful. 

https://docs.cosmos.network/v0.46/building-modules/module-manager.html#appmodule says:

```
	// Deprecated: use RegisterServices
	QuerierRoute() string
```

> QuerierRoute() (deprecated): Returns the name of the module's query route, for [queries](https://docs.cosmos.network/v0.46/building-modules/messages-and-queries.html#queries) to be routes to the module by [BaseApp](https://docs.cosmos.network/v0.46/core/baseapp.html#query-routing).

and this separate page: https://docs.cosmos.network/v0.46/building-modules/messages-and-queries.html#queries which makes me think `QuerierRoute` is only relevant if the file has defined a `query.proto`. Since the minfee module doesn't have any queries, we likely don't need it but I don't see the harm in returning the module name.